### ISSUE TITLE
Allow double speed when watching a lecture for attendance

### ISSUE THREAD

**Issue by elihschiff** [rcos/Venue2#135]
Hi,
I watch all videos at 2x+ speed and have been doing that for years. This includes youtube videos, tv shows, lectures, etc. I am fully able to comprehend videos up to 4x speed. While I understand not wanting to let people cheat the system and skip ahead, I believe that 2x speed is a very reasonable compromise that people can easily comprehend lectures at. I personally get more out of the lecture if it is 2x+ speed vs normal speed. This is because it keeps my mind from wandering as much, and I am sure I am not the only one that feels this way.

Many major video platforms that are used for lectures allow faster playback speed including webex, youtube, box, etc.

I am requesting that you consider bumping up the max video speed that we are allowed to play videos be upped to 2x the normal speed of the video. (you do not even have to make this an official option in the video player, there are Firefox plugins to speed up videos.)

I would also be happy to make a PR to change this if it will be approved.

Thanks for your consideration!

**Comment by TheStopsign** [comment]
Hi Eli,

I'm intrigued by the feature request.
First, may I verify that you tried to use a plugin, and that the plugin wouldn't work? If so, I may already know where the bottleneck/preventing code is.
The next time we (Venue) talk as a team, we will broach the topic.

Personally, I'm on the fence. I understand that some students may learn better at 2x+ speed, as they are able to focus/filter out useless information better, etc, and I actually know quite a few people who feel the same way. However, I'm not sure if this fills Venue's use case. Ideally, classes aren't taught online, and you would be attending lecture in-person. The whole playback idea was specifically for students in emergency situations, or out of the lecture host's time zone. I suspect students (once we were all back online) may be inclined to try to skip class and just speed through lecture.

On the other hand, we do have the option for professors to add polls to lectures ('preventing' students from 'afk'-ing) so 2x+ speed isn't the most unreasonable request, but at the same time I feel as though it could be abused, and after all, Venue's motto/slogan/statement/whathaveyou is "secure attendance."

These are just my own personal initial thoughts. We won't count out the idea just yet. As I said, I'm intrigued by the idea. Thanks for the suggestion! Open Source best Source

**Comment by elihschiff** [comment]
Yes I did try a plugin and it did not work. I quickly glanced at your codebase and I think this might be the line causing the issue, but I could be wrong. https://github.com/rcos/Venue2/blob/d9d2790eaad894b8b45a9faf3ac1dde15cef5ac2/src/components/RestrictedPlayback.vue#L68

here is the plugin I use https://addons.mozilla.org/en-US/firefox/addon/videospeed/

Additionally, I am pretty sure one can just mess with my system clock to get double speed playback to work anyway so I am not sure the "secure attendance" motto would really stop anyone. Therefore I think it would be a nice feature for people not to have to do that if they want to play videos at 2x speed and just allow it on the website to begin with.

**Comment by TheStopsign** [comment]
Hey @elihschiff , we discussed you feature proposal, just wanted to touch back with you as to our plan. And real quick, just to address your system clock comment, we do already have plans to shift to server-side time checking, as well as checking the time on submission. So, that is a TODO item.

More specifically to the feature request: So after some internal discussion, as well as with teachers both affiliated and not affiliated with RPI, we've realized that there is some ground to your proposal. However, we don't believe that this should be a widespread ability. We _are_ going to create this as an optional feature, but we are going to leave that decision to the instructors when it comes to watching playback with restrictions on either a per course, or per lecture basis (undetermined which as of now). For unrestricted playback, this is totally fine, and we are going to get to that.

For now though, we are in the process of implementing CI (Unit/E2E) testing for priority 1. We also have a few functionality kinks to work out on the instructor side, but we WILL get to this.

Thank you again for the ideas! Students' views are just as important! (This is primarily a classroom tool, after all) If you have any other ideas, please reach out again!
